diff --git a/util/archutil/detect.go b/util/archutil/detect.go
index 7826441..5808333 100644
--- a/util/archutil/detect.go
+++ b/util/archutil/detect.go
@@ -78,6 +78,11 @@ func SupportedPlatforms(noCache bool) []ocispecs.Platform {
 			arr = append(arr, linux(p))
 		}
 	}
+	if p := "loong64"; def.Architecture != p {
+		if _, err := loong64Supported(); err == nil {
+			arr = append(arr, linux(p))
+		}
+	}
 	if p := "arm"; def.Architecture != p {
 		if _, err := armSupported(); err == nil {
 			p := linux("arm")
@@ -144,6 +149,11 @@ func WarnIfUnsupported(pfs []ocispecs.Platform) {
 					printPlatformWarning(p, err)
 				}
 			}
+			if p.Architecture == "loong64" {
+				if _, err := loong64Supported(); err != nil {
+					printPlatformWarning(p, err)
+				}
+			}
 			if p.Architecture == "arm" {
 				if _, err := armSupported(); err != nil {
 					printPlatformWarning(p, err)
